array_api_compat/common/_typing.py

```diff
@@ -70,72 +69,11 @@ def shape(self, /) -> _T_co: ...
 DTypeKind: TypeAlias = _DTypeKind | tuple[_DTypeKind, ...]
 
 
-# `__array_namespace_info__.dtypes(kind="bool")`
```

Comment:
holy overengineering Batman!

Reply:
I do not see much benefit at all in breaking down which dict keys each of the methods of `dtypes()` can return. From both a maintainer and a end user perspective, I can see very little to no gain for any work in addition to `-> Mapping[str, DType]`.

Reply:
Two comments above, I explain how you can determine the type of e.g. boolean and integer dtypes, and how this batmobile is the (only) way of getting there. With this, it becomes possible to annotate an array-api function that rejects boolean arrays, yet accepts integer arrays.

Reply:
I don't have the bandwidth to discuss this anymore. Reverting.